### PR TITLE
Add safety checks to IPC (tests)

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/RPCSession.cpp [hunk 1]
```diff
@@ -187,7 +188,7 @@ string RPCSession::eth_getStorageRoot(string const& _address, string const& _blo
 
 void RPCSession::personal_unlockAccount(string const& _address, string const& _password, int _duration)
 {
-	rpcCall("personal_unlockAccount", { quote(_address), quote(_password), to_string(_duration) });
+	BOOST_CHECK(rpcCall("personal_unlockAccount", { quote(_address), quote(_password), to_string(_duration) }) == true);
```
Comment (chriseth) [inline test/RPCSession.cpp]: Is the `== true` required?
Furthermore, if this call does not work, it does not make much sense in continuing the test, does it? So `BOOST_REQUIRE` might be better here.

**Review by chriseth — CHANGES_REQUESTED** [review]

File: test/RPCSession.cpp (hunk 1 above)
Comment (axic) [reply test/RPCSession.cpp]: Isn't `BOOST_REQUIRE(x)` == `BOOST_CHECK(x == true)`?

I don't mind which one is used. Only choose check to make it clear what is the expected return value of each RPC call. (And require was used by the other instance of JSON.)

Comment (chriseth) [reply test/RPCSession.cpp]: no, both of them basically do `== true`. The difference is that require leads to a termination of the test case, check continues.

**Review by chriseth — APPROVED** [review]

File: test/RPCSession.cpp [hunk 2]
```diff
@@ -109,7 +111,6 @@ string IPCSocket::sendRequest(string const& _req)
 	if (!fSuccess)
 		BOOST_FAIL("ReadFile from pipe failed");
 
-	cerr << ".";  //Output for log activity
```
Comment (chriseth) [inline test/RPCSession.cpp:112]: I think this might be needed for appveyor.

**Review by chriseth — APPROVED** [review]

**Pull request merged**